### PR TITLE
Quadrupedal environment

### PR DESCRIPTION
**Pull request by HaojieSHI98**
1st pr for Quadrupedal environment

### CODE REVIEW

File: rlschool/A1_robot/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+```
+## Quick Start
+Quadrupedal environment follows the standard gym APIs to create, run and close an environment.
+
+```python
+# We show a simple example to start Quadrupedal here
+from rlschool import make_env
+import numpy as np
+env = make_env('Quadrupedal',render=1,task="stairstair")
+observation,info = env.reset()
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:29]: The original gym API does return `info`.

File: rlschool/A1_robot/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+| :----------------------:| :-----: | :----------------------------------------: |
+| Distance                   | 3   | The movement of body                 |
+| Motor Angle            | 12     | The current motor angle                |
+| Motor Angle ACC                | 12   | The accelerate of motor angle         |
+| IMU            | 6   | Yaw, Pitch, Roll, d_Yaw, d_Pitch, d_Roll              |
+| Contact               | 4     | The foot contact information          |
+
+## Reward
+
+The simple reward is the movement of the body along the x axis.
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:90]: We want the quadrupedal robot to traverse the challenging scenarios. The reward function depends on the movement speed of the body along the x axis.

File: rlschool/A1_robot/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+## Action
+
+Our action a(t) is a 12 dim vector, corresponding to the desired joint angle of 12 joints. The desired joint angle will be sent to a low level PD controller to control the motor.
+
+We have defined a default joint angle as J_default = [0,0.9,-1.8,0,0.9,-1.8,0,0.9,-1.8,0,0.9,-1.8], the final output will be J_final = a(t) + J_default.
+
+## Observation
+
+| Name                    | Dim    | Description                                |
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:80]: `Dim` -> `Dimension`

File: rlschool/A1_robot/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+| :----------------------:| :-----: | :----------------------------------------: |
+| Distance                   | 3   | The movement of body                 |
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:82]: Distance -> Speed.
The speed of the robot at the global coordinate.

File: rlschool/A1_robot/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+| Motor Angle            | 12     | The current motor angle                |
+| Motor Angle ACC                | 12   | The accelerate of motor angle         |
+| IMU            | 6   | Yaw, Pitch, Roll, d_Yaw, d_Pitch, d_Roll              |
+| Contact               | 4     | The foot contact information          |
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:86]: Four bool variables indicating if the whether foot is touching the ground.

File: rlschool/A1_robot/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+"slopestair": Climb up slopes and down stairs
+
+"slopeslope": Climb up and down slopes
+
+(Above stair height is 8 cm and width is 25 cm, slope is around 20 degree)
+
+"highstair": Climb up stairs with stair height of 13 cm
+
+"heightfield": Walk over heightfield
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:49]: heightfield -> terrain.

File: rlschool/A1_robot/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+for i in range(100):
+    action = np.random.uniform(-0.3,0.3,size=12)
+    next_obs, reward, done, info = env.step(action)
+```
+## Specify a Task
+There are nine predefined tasks in this environment.
+
+"stairstair": Climb up and down stairs
+
+"stairslope": Climb up stairs and down slopes
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:39]: climb up the stair and down the slope.
The following few lines have the same error. Please correct them.

File: rlschool/A1_robot/README.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+"cliff": Gallop over big gaps with width of 50 cm
+
+"Cave": Crawl into caves with height of 18 cm
+
+<img src="nine_block.gif" width="1000"/>
+
+To specify a task, we can change the value of task in make_env function, for example to select "balancebeam"
+
+```python
+# We show a simple example to start LiftSim here
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:62]: remove this comment.

File: rlschool/A1_robot/README.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+from rlschool import make_env
+import numpy as np
+env = make_env('Quadrupedal',render=1,task="balancebeam")
+observation,info = env.reset()
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:66]: remove the following lines.

File: rlschool/A1_robot/README.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+"Cave": Crawl into caves with height of 18 cm
+
+<img src="nine_block.gif" width="1000"/>
+
+To specify a task, we can change the value of task in make_env function, for example to select "balancebeam"
+
+```python
+# We show a simple example to start LiftSim here
+from rlschool import make_env
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:63]: `import rl_school`.

File: rlschool/A1_robot/README.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+import numpy as np
+env = make_env('Quadrupedal',render=1,task="balancebeam")
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:65]: `rl_school.make_env`

File: rlschool/A1_robot/README.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+for i in range(100):
+    action = np.random.uniform(-0.3,0.3,size=12)
+    next_obs, reward, done, info = env.step(action)
+```
+
+## Action
+
+Our action a(t) is a 12 dim vector, corresponding to the desired joint angle of 12 joints. The desired joint angle will be sent to a low level PD controller to control the motor.
+
+We have defined a default joint angle as J_default = [0,0.9,-1.8,0,0.9,-1.8,0,0.9,-1.8,0,0.9,-1.8], the final output will be J_final = a(t) + J_default.
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/README.md:76]: Note that we have XXXX.

File: rlschool/A1_robot/__init__.py [hunk 13]
```diff
@@ -0,0 +1,17 @@
+"""Set up gym interface for locomotion environments."""
+from rlschool.A1_robot.envs import *
+# import gym
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/__init__.py:3]: remove the following commented lines.

File: rlschool/A1_robot/envs/env_builder.py [hunk 14]
```diff
@@ -0,0 +1,112 @@
+# coding=utf-8
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/envs/env_builder.py:1]: What's the difference between this file and `env_builder_old.py` ?
Remove the copyright notice here. For the copyright notice of the third-party code, please refer to this file: 
https://github.com/PaddlePaddle/PARL/blob/369da17e23812131d963c2e816d340ba2fc089fa/examples/ES/noise.py#L1

File: setup.py [hunk 15]
```diff
@@ -54,6 +54,11 @@
         'colour>=0.1.5',
         'scipy>=0.12.0',
         'gym==0.18.0',
+        'pybullet==3.0.7',
+        'attrs==20.3.0',
+        'numba==0.52.0',
+        'filterpy==1.4.5',
+        'absl-py==0.11.0',
```
Comment (TomorrowIsAnOtherDay) [inline setup.py:61]: remove this unnecessary dependency.

File: rlschool/A1_robot/robots/a1_robot_velocity_estimator.py [hunk 16]
```diff
@@ -0,0 +1,106 @@
+"""Estimates base velocity for A1 robot from accelerometer readings."""
```
Comment (TomorrowIsAnOtherDay) [inline rlschool/A1_robot/robots/a1_robot_velocity_estimator.py:1]: I do not see any file using this estimator. 
Is it necessary for the simulator?

File: rlschool/A1_robot/README.md (hunk 1 above)
Comment (HaojieSHI98) [reply rlschool/A1_robot/README.md:29]: ?

File: setup.py (hunk 15 above)
Comment (HaojieSHI98) [reply setup.py:61]: attrs还是必要的，其余的都删掉了

**Comment by WorldEditors** [comment]
Does it include the basic task of "Walking on A Flat Ground"? I think this is needed

**Comment by HaojieSHI98** [comment]
> Does it include the basic task of "Walking on A Flat Ground"? I think this is needed

I have updated it and add the task "Walking on A Flat Ground".

**Pull request merged**